ui/src/app/views/bug-report/component.tsx

```diff
@@ -0,0 +1,139 @@
+import {Component, Input} from '@angular/core';
+
+import {bugReportApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withUserProfile} from 'app/utils/index';
+
+import {AlertDanger} from 'app/components/alert';
+import {Button} from 'app/components/buttons';
+import {ClrIcon} from 'app/components/icons';
+import {Modal, ModalBody, ModalFooter, ModalInput, ModalTitle} from 'app/components/modals';
+import {
+  BugReport,
+  BugReportType,
+  Profile
+} from 'generated/fetch';
+import * as React from 'react';
+
+
+const styles = reactStyles({
+  fieldHeader: {
+    fontSize: 14,
+    color: '#262262',
+    fontWeight: 200,
+    display: 'block',
+    marginTop: '1rem'
+  },
+  field: {
+    fontWeight: 400,
+    width: '100%'
+  }
+});
+
+export const BugReportModal = withUserProfile()
+(class extends React.Component<
+    {profileState: { profile: Profile, reload: Function },
+      bugReportDescription: string, onClose: Function},
+    {bugReport: BugReport, submitting: boolean,
+      sendBugReportError: boolean}> {
+
+  private emptyBugReport = {
+    bugType: BugReportType.APPLICATION,
+    shortDescription: '',
+    reproSteps: '',
+    includeNotebookLogs: true,
+    contactEmail: ''
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      submitting: false,
+      sendBugReportError: false,
+      bugReport: {...this.emptyBugReport,
+        shortDescription: props.bugReportDescription || '',
+        contactEmail: props.profileState.profile.username
+      }
+    };
+  }
+
+  async send() {
```

Comment:
nit: Using `async` here suggests that the function will use `await` on promises, but it doesn't.